[Dallas] Enable interrupts when no sensor found

TD-er opened the pull request:
Fixes: #1833

TD-er committed:
[Dallas] Enable interrupts when no sensor found

tonhuisman reviewed (commented):
That isn't going to compile, as ESP32noInterrupts()/ESP32interrupts() are macros that expand into adding a set of curly braces and a mutex around the wrapped code...

TD-er committed:
[Dallas iButton] Enable interrupts when no sensor found

TD-er committed:
[Dallas] Fix NoInterrupt calls for ESP32

TD-er committed:
Uncrustify both Dallas plugins

TD-er commented:
> That isn't going to compile, as ESP32noInterrupts()/ESP32interrupts() are macros that expand into adding a set of curly braces and a mutex around the wrapped code...

I noticed.
I originally changed it in the Github edit. But now I made changes on the laptop + compile.

So this should work.
Please note the last commit is a code cleanup (uncrustify), so don't view the entire change in this PR to see what has changed.

TD-er committed:
[Dallas] Make sure function forward declaration is on one line

Arduino code may not recognize a forward declaration when it is split over multiple lines.

TD-er committed:
[Dallas] Add warning about the need for a pull up

uzi18 commented on src/_P004_Dallas.ino:
```diff
@@ -77,6 +77,8 @@ boolean Plugin_004(byte function, struct EventStruct *event, String& string)
     case PLUGIN_WEBFORM_LOAD:
     {
+      addFormNote(F("External pull up resistor is needed, see docs!"));
+
```
@TD-er should be more like: 
```
if (Plugin_004_DallasPin != -1) {
  if (digitalRead(Plugin_004_DallasPin) != HIGH) {
    addFormNote(F("External pull up resistor is needed, see docs!"));
  }
 ...
```
what do you think?

TD-er replied:
An internal pull up could be active here, which may not be enough as stated in the issue discussion.

But I can add it... We'll see what happens via the issues :)

TD-er commented:
Hmm the check for the pin state is rather confusing.
If you switch the pin, the state is not read, so only after reloading the plugin page it will show the warning.

I will remove the pin state check, as it is rather confusing to be honest.

TD-er committed:
Merge branch 'mega' into bugfix/dallas_esp32
